interfaces/nginx_route/v0/schema.py

```diff
@@ -0,0 +1,149 @@
+        examples=[
+            "SecAction id:900130,phase:1,nolog,pass,t:none,setvar:tx.crs_exclusions_wordpress=1\n"
+        ],
+        title="Owasp Modsecurity Custom Rules",
+    )
+    path_routes: Optional[str] = Field(
+        None,
+        alias="path-routes",
+        description='Comma separated list of the routes under the hostname that you wish to map to the relation. Example: "/admin,/portal" will map example.test/admin and example.test/portal only. ',
+        examples=["/admin,/portal", "/something(/|$)(.*)"],
+        title="Path Routes",
+    )
+    proxy_read_timeout: Optional[int] = Field(
+        None,
+        alias="proxy-read-timeout",
+        description="Defines a timeout in seconds for reading a response from the proxied server.",
+        examples=[],
+        title="Proxy Read Timeout",
+    )
+    retry_errors: Optional[str] = Field(
+        None,
+        alias="retry-errors",
+        description='Specifies in which cases a request should be retried against the next server. Comma-separated list, e.g. "error,timeout,http_502,http_503,http_504". See http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_next_upstream for more details. Unrecognised values will be ignored. The nginx default will be used if this config option is set to an empty value. ',
+        examples=["error,timeout,http_502,http_503,http_504"],
+        title="Retry Errors",
+    )
+    rewrite_enabled: Optional[bool] = Field(
+        None,
+        alias="rewrite-enabled",
+        description="Whether requests should be written to the `rewrite-target`",
+        examples=[],
+        title="Rewrite Enabled",
+    )
+    rewrite_target: Optional[str] = Field(
+        None,
+        alias="rewrite-target",
+        description='The path to rewrite requests to. If not set, rewrite-target will be "/".',
+        examples=["/app1", "/$2"],
+        title="Rewrite Target",
+    )
+    session_cookie_max_age: Optional[int] = Field(
+        None,
+        alias="session-cookie-max-age",
+        description="The max age to configure a session cookie for. Leaving unset or setting to 0 will disable session cookies and cookie-based affinity.",
+        examples=[],
+        title="Session Cookie Max Age",
+    )
+    tls_secret_name: Optional[str] = Field(
+        None,
+        alias="tls-secret-name",
+        description="The name of the TLS secret to use. Leaving this empty will configure an ingress with TLS disabled.",
+        examples=[],
+        title="Tls Secret Name",
+    )
+    whitelist_source_range: Optional[str] = Field(
+        None,
+        alias="whitelist-source-range",
+        description="Allowed client IP source ranges. The value is a comma separated list of CIDRs.",
+        examples=["10.0.0.0/8,172.16.0.0/12,192.168.0.0/16"],
+        title="Whitelist Source Range",
+    )
+
+
+class ProviderSchema(BaseModel):
```

Comment:
ProviderSchema and RequirerSchema should inherit from `interface_tester.schema_base.DataBagSchema`.
The rest looks good.

Don't forget to `tox -e build-json-schemas` and commit the results.

Reply:
Okay, done. I merged the latest changes from main and updated `shema.py` to use `interface_tester.schema_base.DataBagSchema`, and also built the JSON schema.